### ISSUE TITLE
Build error on Ubuntu 16.04.2 LTS

### ISSUE THREAD

**Issue by ksl67** [cpp-netlib/cpp-netlib#826]
The following errors are reported  when I try to build cpp-netlib-0.11.2-final on Ubuntu 16.04.2 LTS with boost 1.60.0 and gcc 5.4.0 20160609.

`
/home/dev/cots/open-ssl/lib/libcrypto.a(dso_dlfcn.o): In function `dlfcn_globallookup':
dso_dlfcn.c:(.text+0x11): undefined reference to `dlopen'
dso_dlfcn.c:(.text+0x24): undefined reference to `dlsym'
dso_dlfcn.c:(.text+0x2f): undefined reference to `dlclose'
/home/dev/cots/open-ssl/lib/libcrypto.a(dso_dlfcn.o): In function `dlfcn_bind_func':
dso_dlfcn.c:(.text+0x334): undefined reference to `dlsym'
dso_dlfcn.c:(.text+0x3db): undefined reference to `dlerror'
/home/dev/cots/open-ssl/lib/libcrypto.a(dso_dlfcn.o): In function `dlfcn_bind_var':
dso_dlfcn.c:(.text+0x454): undefined reference to `dlsym'
dso_dlfcn.c:(.text+0x4fb): undefined reference to `dlerror'
/home/dev/cots/open-ssl/lib/libcrypto.a(dso_dlfcn.o): In function `dlfcn_load':
dso_dlfcn.c:(.text+0x569): undefined reference to `dlopen'
dso_dlfcn.c:(.text+0x5cb): undefined reference to `dlclose'
dso_dlfcn.c:(.text+0x603): undefined reference to `dlerror'
/home/dev/cots/open-ssl/lib/libcrypto.a(dso_dlfcn.o): In function `dlfcn_pathbyaddr':
dso_dlfcn.c:(.text+0x69f): undefined reference to `dladdr'
dso_dlfcn.c:(.text+0x709): undefined reference to `dlerror'
/home/dev/cots/open-ssl/lib/libcrypto.a(dso_dlfcn.o): In function `dlfcn_unload':
dso_dlfcn.c:(.text+0x762): undefined reference to `dlclose'
collect2: error: ld returned 1 exit status
libs/network/test/CMakeFiles/cpp-netlib-message_test.dir/build.make:110: recipe for target 'tests/cpp-netlib-message_test' failed
make[2]: *** [tests/cpp-netlib-message_test] Error 1
make[2]: Leaving directory '/home/dev/cots/cpp-netlib/build'
CMakeFiles/Makefile2:230: recipe for target 'libs/network/test/CMakeFiles/cpp-netlib-message_test.dir/all' failed
make[1]: *** [libs/network/test/CMakeFiles/cpp-netlib-message_test.dir/all] Error 2
make[1]: Leaving directory '/home/dev/cots/cpp-netlib/build'
Makefile:141: recipe for target 'all' failed
make: *** [all] Error 2
`

**Comment by carun** [comment]
@ksl67 can use 0.13-release branch unless there is a compelling reason to use 0.11.2 branch? IIRC, lots of bug fixes went into 0.13

**Comment by ksl67** [comment]
0.13-release is no good to me. There are changes to the response, changes to how POST requests are handled (see https://github.com/cpp-netlib/cpp-netlib/issues/823). Very little to non-existent documentation or examples. I'll be dumping this library as soon as I get the chance to schedule looking at alternatives.

**Comment by anonimal** [comment]
>I'll be dumping this library as soon as I get the chance to schedule looking at alternatives.

cpp-netlib development is essentially dead. @glynos is MIA and @deanberris's availability is mostly limited to popping in from time to time to review and merge PRs. Issues are piling up and the occasional contributor only PR's a few lines here and there for unrelated issues.

[Boost.Beast](https://github.com/boostorg/beast) is a promising solution. @vinniefalco is actively developing the library.

**Comment by os12** [comment]
> Boost.Beast is a promising solution.

Yes, it is. Yet that is a lower-level lib - it takes quite a bit of code to implement a simple embedded server (there are ample examples though).

**Comment by deanberris** [comment]
I'm really sorry about the lack of availability on this @ksl67.

I'd appreciate your candid feedback on how we can do better aside from being more active on the development of the library.
